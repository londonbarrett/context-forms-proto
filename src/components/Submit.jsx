import * as React from 'react';
import styled from 'styled-components';
import { withFormContext } from './Forms';

const Input = styled.input`
  background: ${({ disabled }) => (disabled ? '#AFDBD7' : '#00544C')};
  border: none;
  border-radius: 1rem;
  color: ${({ disabled }) => (disabled ? '#7DB2AD' : '#FFF')};
  font-size: 1rem;
  line-height: 2rem;
  padding: .5rem 2rem;
`;

const Submit = ({ hasErrors, ...props }) => (
  <Input
    type="submit"
    disabled={hasErrors}
    {...props}
  />
);

export default withFormContext(Submit);
