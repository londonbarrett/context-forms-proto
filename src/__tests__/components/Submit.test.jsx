import React from 'react';
import { render } from 'enzyme';
import Submit from '../../components/Submit';

it('Renders correctly', () => {
  const wrapper = render(<Submit />);
  expect(wrapper).toMatchSnapshot();
});
