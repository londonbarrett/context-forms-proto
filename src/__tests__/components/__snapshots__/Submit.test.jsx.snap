// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<input
  class="sc-bdVaJa bghzgq"
  type="submit"
>
  
</input>
`;
